core/src/main/java/tc/oc/pgm/stats/StatsModule.java

```diff
@@ -0,0 +1,90 @@
+import tc.oc.pgm.api.module.exception.ModuleLoadException;
+import tc.oc.pgm.api.player.MatchPlayer;
+import tc.oc.pgm.util.xml.InvalidXMLException;
+import tc.oc.pgm.util.xml.XMLFluentParser;
+import tc.oc.pgm.util.xml.XMLUtils;
+import tc.oc.pgm.variables.Variable;
+import tc.oc.pgm.variables.VariablesMatchModule;
+
+public class StatsModule implements MapModule<StatsMatchModule> {
+  private final List<MapVariableStat> variables;
+
+  private StatsModule(List<MapVariableStat> variables) {
+    this.variables = variables;
+  }
+
+  @Override
+  public @Nullable StatsMatchModule createMatchModule(Match match) throws ModuleLoadException {
+    final List<StatsMatchModule.VariableStat> variableStats = new ArrayList<>();
+    for (MapVariableStat mapVar : variables) {
+      if (mapVar.variable instanceof Variable.TrackedHolders<MatchPlayer>
+          && mapVar.variable instanceof Variable.DefinesDefault<MatchPlayer>) {
+        variableStats.add(new StatsMatchModule.VariableStat(mapVar.variable, mapVar.displayName));
+      }
+    }
+    return new StatsMatchModule(match, variableStats);
+  }
+
+  @Override
+  public Collection<Class<? extends MatchModule>> getSoftDependencies() {
+    return ImmutableList.of(VariablesMatchModule.class);
+  }
+
+  public static class Factory implements MapModuleFactory<StatsModule> {
+    @Override
+    public StatsModule parse(MapFactory factory, Logger logger, Document doc)
+        throws InvalidXMLException {
+      XMLFluentParser parser = factory.getParser();
+      Element statsElement = XMLUtils.getUniqueChild(doc.getRootElement(), "stats");
+      if (statsElement == null) {
+        return new StatsModule(Collections.emptyList());
+      } else {
+        Element additionalStatsElement = XMLUtils.getUniqueChild(statsElement, "additions");
+        List<MapVariableStat> statVars = new ArrayList<>();
+        if (additionalStatsElement != null) {
+          for (Element el : XMLUtils.getChildren(additionalStatsElement, "stat")) {
+            if (el.getAttribute("var") == null) {
+              throw new InvalidXMLException(
+                  "Expected 'var' attribute on custom stat definition", el);
+            }
+            if (el.getAttribute("display-name") == null) {
+              throw new InvalidXMLException(
+                  "Expected 'displayName' attribute on custom stat definition", el);
+            }
+
+            Component displayName = XMLUtils.parseFormattedText(el, "display-name");
+            Variable<MatchPlayer> varDef =
+                parser.variable(el, "var").scope(MatchPlayer.class).required();
+            if (varDef instanceof Variable.TrackedHolders<MatchPlayer>
+                && varDef instanceof Variable.DefinesDefault<MatchPlayer>) {
+              statVars.add(new MapVariableStat(varDef, displayName));
+            } else {
+              throw new InvalidXMLException(
+                  "Expected variable to be a general purpose player-scoped variable", el);
```

Comment:
What does "general purpose" even mean here? you're restricting to literally a single type of variable, dummy variables.
Imagine a mapmaker wants to use xp (gather-the-most-xp kinda map), or health (uhc-like) as the end of the match as the stat, this won't let them for no good reason.

Reply:
Yes it is restricting it to the dummy variable implementation.

> this won't let them for no good reason

Well, without this change they cannot use any variable 🙂
But in all seriousness, I figured we could always expand on this in a follow-up. Though, if you prefer allowing other types of variables i.e. player variables in the initial version that is a totally valid sentiment as well! I'll explore supporting all variable types and see what I can do

Reply:
Its not about supporting different variable types, its about not using variables at all!! Variables are only needed in features that need write access, for read-only access use formulas instead, which is instant support of any kind of variable for the same price

Reply:
Yeah makes sense I get you on using the formulas